application/src/main/java/org/togetherjava/tjbot/commands/moderation/modmail/ModMailCommand.java

```diff
@@ -0,0 +1,184 @@
+
+        guildOption.addChoices(choices);
+
+        getData().addOptions(messageOption, guildOption, anonymousOption);
+
+        modMailChannelNamePredicate =
+                Pattern.compile(config.getModMailChannelPattern()).asMatchPredicate();
+    }
+
+    private Cache<Long, Instant> createCooldownCache() {
+        return Caffeine.newBuilder()
+            .maximumSize(1_000)
+            .expireAfterAccess(COOLDOWN_DURATION_VALUE, TimeUnit.of(COOLDOWN_DURATION_UNIT))
+            .build();
+    }
+
+    @Override
+    public void onSlashCommand(SlashCommandInteractionEvent event) {
+        long userId = event.getUser().getIdLong();
+
+        if (handleIsOnCooldown(userId, event)) {
+            return;
+        }
+        authorToLastModMailInvocation.put(userId, Instant.now());
+        event.deferReply().setEphemeral(true).queue();
+
+        long userGuildId = event.getOption(OPTION_GUILD).getAsLong();
+        Optional<TextChannel> modMailAuditLog = getModMailChannel(event.getJDA(), userGuildId);
+        if (modMailAuditLog.isEmpty()) {
+            logger.warn(
+                    "Cannot find the designated modmail channel in server by id {} with the pattern modmail",
+                    userGuildId);
+            return;
+        }
+
+        MessageCreateAction message = createMessage(event, userId, modMailAuditLog.orElseThrow());
+
+        sendMessage(event, message);
+
+    }
+
+    private boolean handleIsOnCooldown(long userId, SlashCommandInteractionEvent event) {
+        if (isChannelOnCooldown(userId)) {
+            event.reply("Can only be used once per %s minutes.".formatted(COOLDOWN_DURATION_VALUE))
+                .setEphemeral(true)
+                .queue();
+            return true;
+        }
+        return false;
+    }
+
+    private Optional<TextChannel> getModMailChannel(JDA jda, long userGuildI) {
+        return jda.getGuildById(userGuildI)
+            .getTextChannelCache()
+            .stream()
+            .filter(channel -> modMailChannelNamePredicate.test(channel.getName()))
+            .findAny();
+    }
+
+    private MessageCreateAction createMessage(SlashCommandInteractionEvent event, long userId,
+            TextChannel modMailAuditLog) {
+        String userMessage = event.getOption(OPTION_MESSAGE).getAsString();
+        boolean wantsToStayAnonymous = event.getOption(OPTION_STAY_ANONYMOUS).getAsBoolean();
+        String user;
+        MessageCreateAction message;
+        if (wantsToStayAnonymous) {
+            user = "Anonymous";
+            message = modMailAuditLog.sendMessageEmbeds(createModMailMessage(user, userMessage));
+        } else {
+            user = event.getUser().getAsMention();
+            message = modMailAuditLog.sendMessageEmbeds(createModMailMessage(user, userMessage))
+                .addActionRow(DiscordClientAction.General.USER.asLinkButton("Author Profile",
+                        String.valueOf(userId)));
+        }
+        return message;
```

Comment:
there is some duplication here and its a bit hard to read the flow. maybe write:
```java
String user = wantsToStayAnonymous ? "Anonymous" : event.getUser().getAsMention();

MessageCreateAction message = modMailAuditLog.sendMessageEmbeds(
  createModMailMessage(user, userMessage));

if (!wantsToStayAnonymous) {
  message.addActionRow(...);
}

return message;
```

Reply:
Done